### PR TITLE
Initialize taint sync monitor in a separate package

### PR DESCRIPTION
**Pull request by duncanp-sonar**
* avoids reentrant calls on UI thread causing VS to lock up

Relates to #2059

### CODE REVIEW

File: src/Core/Binding/IActiveSolutionBoundTracker.cs [hunk 1]
```diff
@@ -38,6 +38,8 @@ public interface IActiveSolutionBoundTracker
         /// <summary>
         /// Event to notify subscribers when the binding status of a solution have changed.
         /// This occurs when a new solution is opened, or the SonarQube binding status of the solution changes.
+        /// The event is raised after the <see cref="ISonarQubeService"/> connection has finished
```
Comment (duncanp-sonar) [inline src/Core/Binding/IActiveSolutionBoundTracker.cs:41]: Miscellaneous: added documentation to the interface and class to make it clear that the connection has changed by the time the event is raised.

**Comment by duncanp-sonar** [comment]
This change prevents the UI lock-up I was seeing in some circumstances.
I've also tested that the sync monitor can force-create the taint tool window without causing a lock-up (which it will need to do so the tool window caption can be updated dynamically). I'll send a separate PR for that change, and the force-close change if required.

File: src/IssueViz.Security/IssueVizSecurityPackage.cs [hunk 2]
```diff
@@ -36,9 +34,17 @@
 
 namespace SonarLint.VisualStudio.IssueVisualization.Security
 {
+    /*
+        This is a simple package that just provides the basic VS plumbing for the security commands
+        and tool windows.
+
+        It doesn't provide any other services, and doesn't need to be auto-loadable. It will be loaded
+        by VS only when needed i.e. when it has received a reqeust to to invoke a command or display a
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/IssueVizSecurityPackage.cs:42]: reqeust  = typo.

But I don't think this comment is needed, we have other packages without AutoLoad (i.e. IssueVizPackage) so it's not uncommon.

**Review by rita-g-sonarsource — APPROVED** [review]

File: src/IssueViz.Security/IssueVizSecurityPackage.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/IssueVizSecurityPackage.cs]: I don't think we have a clear enough idea what specific packages are for or what there load characteristics are expected to be. I'm not even sure how many packages we have any more. However, my feeling is that we have too many, we don't understand what the relationships between them are, and some of them will be loading unnecessarily/sooner than necessary. This isn't something we can address in the short term. However, I'm going to start adding comments to packages as part of an effort to understand what we currently have.

**Pull request merged**